Scripts/Runtime/Util/Jobs/ParallelCollectionUtil.cs

```diff
@@ -0,0 +1,103 @@
+using System;
+using System.Reflection;
+using Unity.Jobs.LowLevel.Unsafe;
+using UnityEngine;
+
+namespace Anvil.Unity.DOTS.Util
+{
+    /// <summary>
+    /// Utility methods for working with Native Collections in a parallel manner.
+    /// </summary>
+    public static class ParallelCollectionUtil
+    {
+        /// <remarks>
+        /// This is static readonly because Burst will only read from a static variable if it is readonly.
+        /// We are unable to set it directly to <see cref="JobsUtility.JobWorkerMaximumCount"/> because that function
+        /// is an external function and external functions are not allowed to be called from a static constructor which
+        /// is where static readonly variables get assigned.
+        /// We are also unable to lazy instantiate or call <see cref="JobsUtility.JobWorkerMaximumCount"/> within the
+        /// functions in this class that use it because an external function can only be called from the main thread
+        /// and these functions are intended for use inside bursted jobs that run on many different worker threads.
+        ///
+        /// To avoid this, we have a <see cref="RuntimeInitializeOnLoadMethodAttribute"/> on the
+        /// <see cref="ParallelCollectionUtil.Init"/> method which uses reflection to assign the static readonly
+        /// variable.
+        ///
+        /// While a bit hacky, it is ensured to run before any potential jobs execute. The external call is allowed
+        /// and we get the correct value and inject it into the static readonly field. Subsequent bursted jobs are then
+        /// allowed to access it.
+        /// </remarks>
+        private static readonly int JOB_WORKER_MAXIMUM_COUNT = -1;
```

Comment:
Can you add some assertions that this value was actually set? Easier than tracking down the resulting bug if/when this reflected set doesn't work in the future.

Reply:
Welp, turns out that if Burst is enabled this doesn't work at all. 

Had to change it around to use a `SharedStatic` as per the Burst docs. But yes, I added an assert to catch if it's not set.